pkg/tsdb/v3iotsdb_integration_test.go

```diff
@@ -619,6 +619,139 @@ func TestDeleteTSDB(t *testing.T) {
+			deleteFrom:     0,
+			deleteTo:       1599999999999,
+			ignorErrors:	true,
+			data:     		[]tsdbtest.DataPoint{{Time: 1538580053000, Value: 222.2},
+				{Time: 1538925653000, Value: 333.3},
+				{Time: 1539271253000, Value: 444.4}},
+			expected:		[]tsdbtest.DataPoint{},
+			expectFail: 	false,
+		},
+		{desc: "Should skip partial partition at begining",
+			deleteFrom:     1538580050000,
+			deleteTo:       1999271259000,
+			ignorErrors:	true,
+			data:     		[]tsdbtest.DataPoint{{Time: 1538580053000, Value: 222.2},
+				{Time: 1538925653000, Value: 333.3},
+				{Time: 1539271253000, Value: 444.4}},
+			expected:     	[]tsdbtest.DataPoint{{Time: 1538580053000, Value: 222.2}},
+			expectFail: 	false,
+		},
+		{desc: "Should skip partial partition at end",
+			deleteFrom:     0,
+			deleteTo:       1539271259000,
+			ignorErrors:	true,
+			data:     		[]tsdbtest.DataPoint{{Time: 1538580053000, Value: 222.2},
+				{Time: 1538925653000, Value: 333.3},
+				{Time: 1539271253000, Value: 444.4}},
+			expected:     	[]tsdbtest.DataPoint{{Time: 1539271253000, Value: 444.4}},
+			expectFail: 	false,
+		},
+		{desc: "Should skip partial partition at beginning and end not in range",
+			deleteFrom:     1538580054000,
+			deleteTo:       1539271252000,
+			ignorErrors:	true,
+			data:     		[]tsdbtest.DataPoint{{Time: 1538580053000, Value: 222.2},
+				{Time: 1538925653000, Value: 333.3},
+				{Time: 1539271253000, Value: 444.4}},
+			expected:     	[]tsdbtest.DataPoint{{Time: 1538580053000, Value: 222.2},
+				{Time: 1539271253000, Value: 444.4}},
+			expectFail: 	false,
+		},
+		{desc: "Should skip partial partition at beginning and end although in range",
+			deleteFrom:     1538580050000,
+			deleteTo:       1539271259000,
+			ignorErrors:	true,
+			data:     		[]tsdbtest.DataPoint{{Time: 1538580053000, Value: 222.2},
+				{Time: 1538925653000, Value: 333.3},
+				{Time: 1539271253000, Value: 444.4}},
+			expected:     	[]tsdbtest.DataPoint{{Time: 1538580053000, Value: 222.2},
+				{Time: 1539271253000, Value: 444.4}},
+			expectFail: 	false,
+		},
+	}
+
+	for _, test := range testCases {
+		t.Run(test.desc, func(t *testing.T) {
+			if test.ignoreReason != "" {
+				t.Skip(test.ignoreReason)
+			}
+			testDeleteTSDBCase(t, v3ioConfig, "metricToDelete", utils.LabelsFromStrings("os","linux"),
+				test.data, test.deleteFrom, test.deleteTo, test.ignorErrors, test.expected, test.expectFail)
+		})
+	}
+}
+
+
+func testDeleteTSDBCase(test *testing.T, v3ioConfig *config.V3ioConfig, metricsName string, userLabels []utils.Label,
+	data []tsdbtest.DataPoint, deleteFrom int64, deleteTo int64, ignoreErrors bool,
+	expected []tsdbtest.DataPoint, expectFail bool) {
+
+
+	adapter, teardown := tsdbtest.SetUpWithData(test, v3ioConfig, metricsName, data, userLabels)
+	defer teardown()
+
+	if err := adapter.DeleteDB(false, ignoreErrors, deleteFrom, deleteTo); err != nil {
+		test.Fatalf("Failed to delete DB on teardown. reason: %s", err)
+	}
+
```

Comment:
Check also the number of partitions remaining after partial delete.
For example:
```
schema := testutils.CreateSchema(test, "")
pm, err := partmgr.NewPartitionMngr(schema, nil, v3ioConfig)

initiaPartitions := pm.PartsForRange(0, time.Now().Unix() * 1000, false)
initialNumberOfPartitions := len(initiaPartitions)
	
partitionsToDelete := pm.PartsForRange(deleteFrom, deleteTo, false)
	
assert.Equal(test, len(partitionsToDelete), numberOfPartitionsInRange)
	
if err := adapter.DeleteDB(false, ignoreErrors, deleteFrom, deleteTo); err != nil {
	test.Fatalf("Failed to delete DB on teardown. reason: %s", err)
}

remainingParts := pm.PartsForRange(0, time.Now().Unix() * 1000, false)
assert.Equal(test, remainingParts,  initialNumberOfPartitions - numberOfPartitionsInRange)
```